overlord/configstate/transaction_test.go

```diff
@@ -165,3 +166,27 @@ func (s *transactionSuite) TestIsolationFromOtherTransactions(c *C) {
 	c.Check(transaction2.Get("test-snap", "foo", &value), IsNil)
 	c.Check(value, Equals, "initial", Commentf("Expected transaction2 to be isolated from transaction1"))
 }
+
+type brokenType struct {
+	on string
+}
+
+func (b *brokenType) UnmarshalJSON(data []byte) error {
+	if b.on == string(data) {
+		return fmt.Errorf("BAM!")
+	}
+	return nil
+}
+
+func (s *transactionSuite) TestGetUnmarshalError(c *C) {
```

Comment:
Thanks, I've actually improved it a bit further locally, to catch the error both on the pristine state and in the changed state.